File: sdk/swift/Sources/TelecomSDK/RatingPlanAPI.swift
import Foundation

/// API for rating plan management
public class RatingPlanAPI {
    private let client: HTTPClient
    
    public init(client: HTTPClient) {
        self.client = client
    }
    
    /// List all available rating plans
    public func list() async throws -> [RatingPlan] {
        return try await client.get(path: "/v1/rating-plans")
    }
    
    /// Get a rating plan by ID
    public func get(planID: String) async throws -> RatingPlan {
        return try await client.get(path: "/v1/rating-plans/\(planID)")
    }
}

File: sdk/kotlin/src/main/kotlin/com/telecom/RatingPlanAPI.kt
package com.telecom

/**
 * API for rating plan management
 */
class RatingPlanAPI(private val client: HTTPClient) {
    
    /**
     * List all available rating plans
     */
    suspend fun list(): List<RatingPlan> {
        return client.get("/v1/rating-plans")
    }
    
    /**
     * Get a rating plan by ID
     */
    suspend fun get(planId: String): RatingPlan {
        return client.get("/v1/rating-plans/$planId")
    }
}
